### ISSUE TITLE
move logic from shell/gulpfile.js to sigh

### ISSUE THREAD

**Issue by smalls** [PolymerLabs/arcs#828]
Most particularly the webpack logic.

Also ensure that this build examines the inputs before deciding to run, to avoid expensive (and duplicate) work.

**Comment by mykmartin** [comment]
Matt: do you still think the second item (gate builds on inputs) is important? I assume that was intended to be something along the lines of avoiding rebuilds when the inputs weren't updated. If so, I don't think that would currently be worth the effort required.

**Comment by smalls** [comment]
I think it will be important at some point, but I totally buy that it's not yet and therefore not worth doing.

I'm generally worried about the code-build-test cycle getting too long (particularly as we move ahead integrating more testing into sigh, including the selenium tests which take a while), but there are other mitigations (#827, or expanding the watch functionality) that would be as effective as gating the builds on inputs.

So long story short I think we can close this out with your changes?
